src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_MAX_SIZE = 10;
+    private static final int FIGURE_MAX_PARAMETERS_COUNT = 3;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    Figure getRandomFigure() {
+        int[] figureParameters = new int[FIGURE_MAX_PARAMETERS_COUNT];
+        for (int i = 0; i < FIGURE_MAX_PARAMETERS_COUNT; i++) {
+            figureParameters[i] = random.nextInt(FIGURE_MAX_SIZE) + 1;
+        }
+        String figureColorRandom = colorSupplier.getRandomColor();
+        FigureTypeName figureTypeRandom = FigureTypeName
+                .values()[random.nextInt(FigureTypeName.values().length)];
+        switch (figureTypeRandom) {
+            case circle:
+                return new Circle(figureColorRandom,
```

Comment:
create a separate method to create each figure and invoke it here

Reply:
For example, will create getRandomCircle,  getRandomSquare, ... and invoke it in FigureSupplier.getRandomFigure ?